packages/asset-swapper/src/utils/market_operation_utils/comparison_price.ts

```diff
@@ -0,0 +1,73 @@
+import { Web3Wrapper } from '@0x/dev-utils';
+import { BigNumber } from '@0x/utils';
+import * as _ from 'lodash';
+
+import { MarketOperation } from '../../types';
+
+import { COMPARISON_PRICE_DECIMALS, SOURCE_FLAGS } from './constants';
+import { ComparisonPrice, MarketSideLiquidity, OptimizerResult } from './types';
+
+/**
+ * Takes in an optimizer response and returns a price for RFQT MMs to beat
+ * returns the price of the taker asset in terms of the maker asset
+ * So the RFQT MM should aim for a higher price
+ * @param optimizerResult the output of the optimizer, contains the best orders
+ * @param amount the amount specified by the client
+ * @param marketSideLiquidity the results from querying liquidity sources
+ * @return ComparisonPrice object with the prices for RFQ MMs to beat
+ */
+export function getComparisonPrices(
+    optimizerResult: OptimizerResult,
+    amount: BigNumber,
```

Comment:
Does `amount` already remove the penalty from all the other sources that make the optimized on-chain quote?